mlrun/api/db/sqldb/db.py

```diff
@@ -266,12 +269,28 @@ def list_functions(
         #     uid = self._resolve_tag(Function, project, uid)
 
         funcs = FunctionList()
-        funcs.extend(
-            obj.struct
-            for obj in self._find_functions(session, name, project, uid, labels)
-        )
+        for obj in self._find_functions(session, name, project, uid, labels):
+            function_dict = obj.struct
+            if not tag:
+                function_tags = self._list_function_tags(session, project, obj.id)
+                for function_tag in function_tags:
+                    function_dict_copy = deepcopy(function_dict)
```

Comment:
@Hedingber do we need to copy ?

Reply:
Talked offline, bigger refactor is coming